lib/super_digit.rb

```diff
@@ -1,15 +1,15 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) where n is the number of digits. as the number doubles, worst case, length of number also doubles. 5*2=10
+# Space Complexity - O(1)
 def super_digit(n)
-   
+  return n if n < 10
+  super_digit(n.digits.sum)
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(1) I'm not sure with these space complexities. methods like .split or .digit... are they O(n)?
 def refined_super_digit(n, k)
```

Comment:
This does work, but it's O( (nk)^2) in time complexity and O(nk) in space. 

If you use dynamic programming, you can reduce this a bit.